domain-concept/src/main/java/life/qbic/domain/concepts/DomainEvent.java

```diff
@@ -13,10 +14,21 @@
  */
 public abstract class DomainEvent implements Serializable {
 
+  protected final Instant occurredOn;
+
+  protected DomainEvent() {
+    this.occurredOn = Instant.now();
+  }
+
```

Comment:
It IS only fired in the process of instantiation, when super() is called. Since it is used in dozens of subclasses (every one), I think this provides both clarity, as well as ease of refactoring. Every event has a timestamp when it occured.